File: math/math_game.pl
#!/usr/bin/perl

use warnings;
use strict;
use CGI qw(:standard);

my $min = param("min");
my $max = param("max") + $min;
my $num_questions = param("questions");

start();
print h2("The Math Game");
# start the form and the table
print start_form("get","math_score.pl",&CGI::URL_ENCODED);
print '<table border="0">',"\n";

my @p;

for(my $i = 1; $i <= $num_questions; $i++) {
  # create and print the problem
  $p[$i] = create_problem($min,$max);
  print '<tr><td>',$p[$i],
        '</td><td>=</td><td>',
        '<input type="text" name="a',$i,'">',
        '</td></tr>',"\n";
}

print "</table>\n";

# put the problems in a hidden file
for(my $i = 1; $i <= $num_questions; $i++) {
print '<input type="hidden" name="p',
      $i,'" value="',$p[$i],'">',"\n";
}

print '<input type="hidden" name="questions"',
      ' value="',$num_questions,'">',"\n";
print '<input type="submit" value="Submit">',"\n";
print end_form;
end();

#################################
# create_problem                #
# Inputs:                       #
#   The minimum and maximum     #
#   values of the numbers to be #
#   added or subtracted         #
# Outputs:                      #
#   A math problem              #
#################################
sub create_problem {
  my $min = shift;
  my $max = shift;
  my $o;
  my $n1 = my_rand($min,$max);
  my $n2 = my_rand($min,$max);
  if (my_rand(0,1) == 0) {
    $o = "-";
    if ($n2 > $n1) {
      my $n3 = $n1;
      $n1 = $n2;
      $n2 = $n3;
    }
  } else {
    $o = "+";
  }
  my $problem = $n1." ".$o." ".$n2;
  return $problem;
}
#################################
# my_rand                       #
# Inputs:                       #
#   The minimum and maximum     #
#   values of the random        #
#   integers                    #
# Outputs:                      #
#   A random number from the    #
#   minimum to the maximum      #
#################################
sub my_rand {
  my $min = shift;
  my $max = shift;
  my $number = $min + int(rand($max - $min + 1));
  return $number;
}
#################################
# start                         #
# Inputs:                       #
#   None                        #
# Outputs:                      #
#   The beginning of a web page #
#################################
sub start {
  print "Content-type: text/html\n\n";
  print `/usr/bin/php ../header.php`;
#  print header(), start_html("The Math Game");
#  print header(), start_html(-title=>"The Math Game", -style =>{-src=>["../stylesheet.css"],-media=> 'all'});
}
#################################
# end                           #
# Inputs:                       #
#   None                        #
# Outputs:                      #
#   The end of a web page       #
#################################
sub end {
  print h6("Made by Johan");
  print `/usr/bin/php ../footer.php`;
#  print end_html();
}

File: math/math_game.php
<?php $pageName = "Math"; include '../header.php'; ?>

<h2>

The Math Game

</h2>

<p>

This game makes addition and subtration math problems, and after you do them, it gives you your score.

</p>

<p>

To adust the difficulty of the game, you can change the maximum and the minimum value of the numbers to be subtracted or added. You can also change the number of questions below.

</p>

<p>

Do not enter text in the answer boxes.

</p>

<form action="math_game.pl">

What do you want the minimum value of the numbers to be subtracted or added to be?

<select name="min">

<option selected="selected">1</option>

<option>2</option>

<option>3</option>

<option>4</option>

<option>5</option>

<option>6</option>

<option>7</option>

<option>8</option>

<option>9</option>

<option>10</option>

<option>11</option>

<option>12</option>

<option>13</option>

<option>14</option>

<option>15</option>

<option>16</option>

<option>17</option>

<option>18</option>

<option>19</option>

<option>20</option>

<option>21</option>

<option>22</option>

<option>23</option>

<option>24</option>

<option>25</option>

<option>26</option>

<option>27</option>

<option>28</option>

<option>29</option>

<option>30</option>

<option>31</option>

<option>32</option>

<option>33</option>

<option>34</option>

<option>35</option>

<option>36</option>

<option>37</option>

<option>38</option>

<option>39</option>

<option>40</option>

<option>41</option>

<option>42</option>

<option>43</option>

<option>44</option>

<option>45</option>

<option>46</option>

<option>47</option>

<option>48</option>

<option>49</option>

<option>50</option>

</select>

<br>

The maximum will be 

<select name="max">

<option>1</option>

<option>2</option>

<option>3</option>

<option>4</option>

<option>5</option>

<option>6</option>

<option>7</option>

<option>8</option>

<option>9</option>

<option selected="selected">10</option>

<option>11</option>

<option>12</option>

<option>13</option>

<option>14</option>

<option>15</option>

<option>16</option>

<option>17</option>

<option>18</option>

<option>19</option>

<option>20</option>

</select>

more than the minimum.

<br>

<br>

How many questions do you want?

<select name="questions">

<option>1</option>

<option>2</option>

<option>3</option>

<option>4</option>

<option>5</option>

<option>6</option>

<option>7</option>

<option>8</option>

<option>9</option>

<option>10</option>

<option>11</option>

<option>12</option>

<option>13</option>

<option>14</option>

<option selected="selected">15</option>

<option>16</option>

<option>17</option>

<option>18</option>

<option>19</option>

<option>20</option>

<option>21</option>

<option>22</option>

<option>23</option>

<option>24</option>

<option>25</option>

<option>26</option>

<option>27</option>

<option>28</option>

<option>29</option>

<option>30</option>

<option>31</option>

<option>32</option>

<option>33</option>

<option>34</option>

<option>35</option>

<option>36</option>

<option>37</option>

<option>38</option>

<option>39</option>

<option>40</option>

<option>41</option>

<option>42</option>

<option>43</option>

<option>44</option>

<option>45</option>

<option>46</option>

<option>47</option>

<option>48</option>

<option>49</option>

<option>50</option>

</select>

<br>

<br>

<input value="Start" type="submit">

</form>

</body>

</html>

<?php include '../footer.php'; ?>
